Chessington.GameEngine/Pieces/King.cs

```diff
@@ -21,7 +21,7 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             {
                 foreach (var colDiff in Enumerable.Range(-1, 3))
                 {
-                    if (Square.At(currentRow + rowDiff, currentCol + colDiff).IsInbound())
+                    if (Square.At(currentRow + rowDiff, currentCol + colDiff).CanMoveOrTake(board, this))
```

Comment:
Classes should have functionalities specific to them and have separation of concerns.
I am not sure that Square class should know about the board.
class "Board" should be able to determine whether it's possible to move into a square or not as it needs to know the state of the board